src/client/Microsoft.Identity.Client/WsTrust/WsTrustWebRequestManager.cs

```diff
@@ -109,16 +113,37 @@ public async Task<MexDocument> GetMexDocumentAsync(string federationMetadataUrl,
         {
             requestContext.Logger.Info("Sending request to userrealm endpoint.");
 
-             var uri = new UriBuilder(userRealmUriPrefix + userName + "?api-version=1.0").Uri;
+            IDictionary<string, string> headers = CreateHttpTelemetryHeaders(requestContext);
 
+            var uri = new UriBuilder(userRealmUriPrefix + userName + "?api-version=1.0").Uri;
+            
             var httpResponse = await _httpManager.SendGetAsync(
                 uri,
-                MsalIdHelper.GetMsalIdParameters(requestContext.Logger),
+                headers,
                 requestContext.Logger).ConfigureAwait(false);
 
             return httpResponse.StatusCode == System.Net.HttpStatusCode.OK
                 ? JsonHelper.DeserializeFromJson<UserRealmDiscoveryResponse>(httpResponse.Body)
                 : null;
         }
+
+        private IDictionary<string, string> CreateHttpTelemetryHeaders(RequestContext requestContext)
+        {
+            IDictionary<string, string> httpTelemetryHeader = new Dictionary<string, string>();
+            httpTelemetryHeader[TelemetryConstants.XClientLastRequest] = requestContext.ServiceBundle.TelemetryManager.FetchAndResetPreviousHttpTelemetryContent();
```

Comment:
An authentication flow involves talking to the several endpoints - discovery, oidc, ws-trust, authroize, token. However, I do not think the server side is ready to accept data on all of these - my understanding is that it only accepts the data on /token (and maybe /authorize?). 

And since you're resetting the last error code everytime you send the telemetry data, we'll most likely lose it. So I propose either: 

1. Only send telemetry on /token endpoint OR
2. Only reset the last error code when sending to /token endpoint